app/Services/AchievementService.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Achievement;
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class AchievementService
+{
+
+    public function createAcievement($data)
+    {
+        return Achievement::query()->create($data);
+    }
+
+    public function updateAchievement($achievementId, $data)
+    {
+        $achievement = Achievement::query()->findOrFail($achievementId);
+        $achievement->update($data);
+        return $achievement;
+    }
+
+    public function deleteAchievement($achievementId)
+    {
+        return Achievement::destroy($achievementId);
+    }
+
+    public function getAllAchievements()
+    {
+        return Achievement::query()->get();
+    }
+
+    public function getAchievementById($achievementId)
+    {
+        return Achievement::query()->findOrFail($achievementId);
+    }
+    public function checkAndAwardAchievements(User $user)
+    {
+
+        if ($user->quizAttempts()->count() > 0) {
+            $this->awardAchievement($user, 'First Quiz Completed');
+        }
+
+        if ($user->quizAttempts()->where('score', '>=', 10)->exists()) {
+            $this->awardAchievement($user, 'High Score');
+        }
```

Comment:
Instead of hardcoding achievement names and conditions, you can use a configuration or database-driven approach to define criteria.

Reply:
Also feel free to experiment and add more achievements and criteria

Reply:
I changed the method, and everything that should be changed, model, service, controller,achievements table with new migration, tests... but instead condition_value to be json type i made it integer beacause as json  i tried everything but it didnt work.